cpp/daal/src/algorithms/kmeans/oneapi/kmeans_lloyd_distr_step1_ucapi_impl.i

```diff
@@ -0,0 +1,241 @@
+    NumericTable * ntObjFunction = const_cast<NumericTable *>(r[2]);
+    NumericTable * ntCValues     = const_cast<NumericTable *>(r[3]);
+    NumericTable * ntCCentroids  = const_cast<NumericTable *>(r[4]);
+    NumericTable * ntAssignments = const_cast<NumericTable *>(r[5]);
+
+    const size_t nIter     = par->maxIterations;
+    const size_t nRows     = ntData->getNumberOfRows();
+    const size_t nFeatures = ntData->getNumberOfColumns();
+    const size_t nClusters = par->nClusters;
+
+    uint32_t blockSize = 0;
+    DAAL_CHECK_STATUS_VAR(this->getBlockSize(nRows, nClusters, nFeatures, blockSize));
+    DAAL_CHECK_STATUS_VAR(this->initializeBuffers(nClusters, nFeatures, blockSize));
+
+    BlockDescriptor<algorithmFPType> inCentroidsRows;
+    DAAL_CHECK_STATUS_VAR(ntInCentroids->getBlockOfRows(0, nClusters, readOnly, inCentroidsRows));
+    auto inCentroids = inCentroidsRows.getBuffer();
+
+    BlockDescriptor<int> ntClusterS0Rows;
+    DAAL_CHECK_STATUS_VAR(ntClusterS0->getBlockOfRows(0, nClusters, writeOnly, ntClusterS0Rows));
+    auto outCCounters = ntClusterS0Rows.getBuffer();
+
+    BlockDescriptor<algorithmFPType> ntClusterS1Rows;
+    DAAL_CHECK_STATUS_VAR(ntClusterS1->getBlockOfRows(0, nClusters, writeOnly, ntClusterS1Rows));
+    auto outCentroids = ntClusterS1Rows.getBuffer();
+
+    BlockDescriptor<algorithmFPType> ntObjFunctionRows;
+    DAAL_CHECK_STATUS_VAR(ntObjFunction->getBlockOfRows(0, nClusters, writeOnly, ntObjFunctionRows));
+    auto outObjFunction = ntObjFunctionRows.getBuffer();
+
+    if (!inCentroids || !outCCounters || !outCentroids || !outObjFunction)
+    {
+        return Status(ErrorNullPtr);
+    }
+
+    BlockDescriptor<algorithmFPType> ntCValuesRows;
+    DAAL_CHECK_STATUS_VAR(ntCValues->getBlockOfRows(0, nClusters, writeOnly, ntCValuesRows));
+    auto outCValues = UniversalBuffer(ntCValuesRows.getBuffer());
+
+    BlockDescriptor<algorithmFPType> ntCCentroidsRows;
+    DAAL_CHECK_STATUS_VAR(ntCCentroids->getBlockOfRows(0, nClusters, writeOnly, ntCCentroidsRows));
+    auto outCCentroids = UniversalBuffer(ntCCentroidsRows.getBuffer());
+
+    context.fill(outCValues, sizeof(algorithmFPType) == 4 ? FLT_MAX : DBL_MAX, &st);
+    DAAL_CHECK_STATUS_VAR(st);
+
+    this->buildProgram(kernelFactory, nClusters, &st);
+    DAAL_CHECK_STATUS_VAR(st);
+
+    auto assignments = context.allocate(TypeIds::id<int>(), blockSize, &st);
+    DAAL_CHECK_STATUS_VAR(st);
+
+    size_t nPartNum = this->getCandidatePartNum(nClusters);
+    size_t nBlocks  = nRows / blockSize + int(nRows % blockSize != 0);
+
+    bool needCandidates = true;
+    for (size_t block = 0; block < nBlocks; block++)
+    {
+        auto range = Range::createFromBlock(block, blockSize, nRows);
+        BlockDescriptor<algorithmFPType> dataRows;
+        DAAL_CHECK_STATUS_VAR(ntData->getBlockOfRows(range.startIndex, range.count, readOnly, dataRows));
+        auto data = dataRows.getBuffer();
+        if (!data)
```

Comment:
As i see in getBlockOfRows implementation data check isn't required after it (getBlockOfRows returns an error in any problem with block creation)